Add CredentialsRequest creation

jsafrane opened the pull request:
Fixes: #11

Create/update CredentialsRequest on sync.

There is no `CredentialsRequest informer for now! It won't get re-created when it's deleted by user!

TODO:
* ~~Delete CredentialsRequest during uninstallation.~~
* ~~Set proper namespace of requested Secret.~~
* ~~Unit tests.~~
* Add CredentialsRequest informer (as a stretch goal).

gnufied commented on pkg/operator/operator_test.go:
```diff
@@ -482,10 +521,42 @@ func TestSync(t *testing.T) {
+					withGenerations(1, 1, 3), // now the operator knows generation 3
+					withTrueConditions(opv1.OperatorStatusTypeAvailable, opv1.OperatorStatusTypeUpgradeable, opv1.OperatorStatusTypePrereqsSatisfied),
+					withFalseConditions(opv1.OperatorStatusTypeDegraded, opv1.OperatorStatusTypeProgressing)),
+				credentialsRequest: getCredentialsRequest(withCredentialsRequestGeneration(1)),
```
minor nit: should we actually verify if operator indeed restores spec field rather than just verifying generation?

jsafrane replied:
Added check for expected / actual `credentialsRequest`.

gnufied commented on pkg/operator/sync.go:
```diff
@@ -192,6 +195,35 @@ func (c *csiDriverOperator) syncRBAC(instance *v1alpha1.EBSCSIDriver) error {
 	return nil
 }
 
+func (c *csiDriverOperator) syncCredentialsRequest(instance *v1alpha1.EBSCSIDriver) (*unstructured.Unstructured, error) {
```
Not a blocker as such - but is it a problem that we create the CredentialRequest but for some reason the secret did not get created and now all our deployments can't start. User will see a cryptic error which won't make sense to them.

jsafrane replied:
Added wait for Secret to appear & sending events for it.

bertinatto commented on pkg/operator/operator.go:
```diff
@@ -302,12 +306,17 @@ func (c *csiDriverOperator) handleSync(instance *v1alpha1.EBSCSIDriver) error {
+	if err != nil {
+		return fmt.Errorf("failed to sync CredentialsRequest: %v", err)
+	}
+
```
I'd preferably create this resource earlier

jsafrane replied:
Moved right after Namespace creation

jsafrane commented:
I added wait for Secret to appear using an informer (I don't want to abuse API server that much).
It takes ~4-8 seconds to get the Secret created, therefore I added `Progressing: true` to be reported while waiting for the Secret or on any other `handleSync()` error, because the operator did not finish what it wanted to do and some progress is expected).

gnufied commented:
/lgtm
